jgrapht-core/src/main/java/org/jgrapht/GraphTests.java

```diff
@@ -495,6 +496,37 @@
         return new ChordalityInspector<>(graph).isChordal();
     }
 
+    /**
+     * Tests whether an undirected graph meets Ore's condition to be Hamiltonian.
+     *
+     * Let G be a (finite and simple) graph with $n \geq 3$ vertices. We denote by deg(v) the degree of a vertex v in G,
+     * i.e. the number of incident edges in G to v.
+     * Then, Ore's theorem states that if $deg(v) + deg(w) \geq n$ for every pair of distinct non-adjacent vertices
+     * v and w of G, then G is Hamiltonian.
+     *
+     * @param graph the input graph
+     * @param <V> the graph vertex type
+     * @param <E> the graph edge type
+     * @return true if the graph meets Ore's condition, false otherwise
+     * @see org.jgrapht.alg.tour.PalmerHamiltonianCycle
+     */
+    public static <V, E> boolean hasOreProperty(Graph<V, E> graph){
+        requireUndirected(graph);
+
+        final int n = graph.vertexSet().size();
+
+        if (!graph.getType().isSimple() || n < 3)
+            return false;
+
+        for (V v: graph.vertexSet())
```

Comment:
You are technically making double the number of required checks (for a pair of vertices u,v, you check both (u,v) and (v,u)). Consider casting the vertex set to an ArrayList to prevent this.